packages/resolve-query/src/facade.js

```diff
@@ -16,42 +33,65 @@ const createFacade = ({ model, gqlSchema, gqlResolvers, customResolvers }) => {
       resolvers: { Query: gqlResolvers }
     })
 
-    Object.defineProperty(executors, 'executeQueryGraphql', {
-      value: async (gqlQuery, gqlVariables, jwtToken) => {
-        const parsedGqlQuery = parse(gqlQuery)
+    const execGraphql = async (gqlQuery, gqlVariables, jwtToken) => {
+      const parsedGqlQuery = parse(gqlQuery)
 
-        const gqlResponse = await execute(
-          executableSchema,
-          parsedGqlQuery,
-          await model(),
-          { jwtToken },
-          gqlVariables
+      const gqlResponse = await execute(
+        executableSchema,
+        parsedGqlQuery,
+        await model(),
+        { jwtToken },
+        gqlVariables
+      )
+
+      if (gqlResponse.errors) throw gqlResponse.errors
+      return gqlResponse.data
+    }
+
+    const makeReactiveReader = async (publisher, ...execGqlArgs) => {
+      if (typeof publisher !== 'function') {
+        throw new Error(
+          'Publisher should be callback function (diff: Object) => void'
         )
+      }
 
-        if (gqlResponse.errors) throw gqlResponse.errors
-        return gqlResponse.data
-      },
-      enumerable: true
-    })
-  }
+      let result = await execGraphql(...execGqlArgs)
 
-  Object.defineProperty(executors, 'dispose', {
-    value: model.dispose.bind(model),
-    enumerable: false
-  })
+      let flowPromise = Promise.resolve()
+      const eventHandler = async () => {
+        if (!flowPromise) return
+        const actualResult = await execGraphql(...execGqlArgs)
+        const difference = diff(result, actualResult)
+        result = actualResult
+        await publisher(difference)
+      }
 
-  Object.defineProperty(executors, 'executeQueryCustom', {
-    value: async (name, ...args) => {
-      if (!customResolvers || !customResolvers[name]) {
-        throw new Error(`The '${name}' custom resolver is not specified`)
+      const eventListener = event =>
+        (flowPromise = flowPromise.then(eventHandler.bind(null, event)))
+      model.addEventListener(eventListener)
+
+      const forceStop = () => {
+        if (!flowPromise) return
+        model.removeEventListener(eventListener)
```

Comment:
Client have some default opportunity to know if removeEventListener happened, don't it?

Reply:
How? There are no desctructors on Node.JS. Since provided callback is a function, which will be called at custom time from promises flow in projection "thread", there must be a way to prevent calling callback after unsubscribe phase.

Reply:
> I think we must notify client about subscription fail.
Of course, client had been notified about any subscription failure. More that, auto re-connection mechanism is implemented on client side, which helps to stay subscription in live mode.

Referent code is about backward thing - to stop internal subscriptions pipeline when current subscription is not more in use. Also, idempotent reaction is guaranteed